tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -1,42 +1,56 @@
+discard """
+  matrix: "--gc:arc --threads:on; --gc:arc --threads:on -d:danger; --threads:on"
+"""
+
 import httpclient, asynchttpserver, asyncdispatch, asyncfutures
 import net
 
 import std/asyncnet
 import std/nativesockets
+import std/threadpool
 
 const postBegin = """
 POST / HTTP/1.1
 Transfer-Encoding:chunked
 
 """
 
-template genTest(input, expected) =
-  var sanity = false
-  proc handler(request: Request) {.async.} =
+template genTest(input, expected: string) =
+  proc handler(request: Request, future: Future[bool]): Future[void] {.async, gcsafe.} =
       doAssert(request.body == expected)
       doAssert(request.headers.hasKey("Transfer-Encoding"))
       doAssert(not request.headers.hasKey("Content-Length"))
-      sanity = true
+      future.complete(true)
       await request.respond(Http200, "Good")
 
-  proc runSleepLoop(server: AsyncHttpServer) {.async.} = 
+  proc runSleepLoop(server: AsyncHttpServer, future: Future[bool]) {.async, gcsafe.} = 
```

Comment:
instead of adding this proc, and calling `syncCheck runSleepLoop(server, handlerFuture)`, how about this (a bit simpler):

```
    proc wrapper2(request: Request): Future[void] {.gcsafe, closure.} =
      handler(request, handlerFuture)
    asyncCheck server.acceptRequest(wrapper2)
```

Reply:
Thanks, this is nicer. Fixed.